### PR TITLE
Replace momentjs with dayjs

### PR DESCRIPTION
**Pull request by lcampos**
Replace momentjs with dayjs. This is the first step into making `@salesforce/core` compatible with bundling tools like webpack, rollup, etc.

@W-6980157@

### CODE REVIEW

File: test/cycles.test.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ test('cycles', function (t) {
             delete o.pid;
             delete o.time;
             // Hack object/dict comparison: JSONify.
-            t.equal(JSON.stringify(o), JSON.stringify(expect[i]),
+            t.equal(JSON.stringify(o).replace(/\s(<ref \*1>|\*1)/gi, ''), JSON.stringify(expect[i]),
```
Comment (lcampos) [inline test/cycles.test.js]: Node 13 adds `*1` when you stringify a json object that has circular references, this removes those from the test comparison.

File: test/log.test.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ test('log.info(undefined, <msg>)', function (t) {
     names.forEach(function (lvl) {
         log3[lvl].call(log3, undefined, 'some message');
         var rec = catcher.records[catcher.records.length - 1];
-        t.equal(rec.msg, 'undefined \'some message\'',
+        t.equal(rec.msg.replace(/\'/gi, ''), 'undefined some message',
```
Comment (lcampos) [inline test/log.test.js]: Node 10 keeps the single quotes while node 12 and latest don't.

File: bin/bunyan [hunk 3]
```diff
@@ -15,7 +15,6 @@ var VERSION = '1.8.2';
 
 var p = console.log;
 var util = require('util');
-var pathlib = require('path');
```
Comment (AnanyaJha) [inline bin/bunyan]: is this not needed anymore?

Comment (lcampos) [reply bin/bunyan]: No, it was never used.

**Pull request merged**